sdk/appcenter/src/test/java/com/microsoft/appcenter/channel/DefaultChannelTest.java

```diff
@@ -1071,4 +1086,164 @@ public void scheduleLogsWithNewToken() {
+        when(mockPersistence.countLogs(TEST_GROUP)).thenReturn(0);
+        Channel.GroupListener mockListener = mock(Channel.GroupListener.class);
+        AppCenterIngestion mockIngestion = mock(AppCenterIngestion.class);
+        DefaultChannel channel = new DefaultChannel(mock(Context.class), UUIDUtils.randomUUID().toString(), mockPersistence, mockIngestion, mAppCenterHandler);
+        channel.addGroup(TEST_GROUP, 10, mockInterval, MAX_PARALLEL_BATCHES, mockIngestion, mockListener);
+
+        /* Check values. */
+        verify(mAppCenterHandler, never()).postDelayed(any(Runnable.class), anyLong());
+        verifyStatic();
+        SharedPreferencesManager.remove(eq(START_TIMER_PREFIX + TEST_GROUP));
+    }
+
+    @Test
+    public void checkPendingLogsPendingLogMoreZero() {
+        long mockInterval = 10000;
+        long mockSystemTime = 50000L;
+        final long mockPrefTime = 1000L;
+
+        /* Mock static classes. */
+        mockStatic(System.class);
+        when(System.currentTimeMillis()).thenReturn(mockSystemTime);
+        when(SharedPreferencesManager.getLong(eq(START_TIMER_PREFIX + TEST_GROUP))).thenReturn(mockPrefTime);
+
+        /* Mock channel. */
+        Persistence mockPersistence = mock(Persistence.class);
+        when(mockPersistence.countLogs(TEST_GROUP)).thenReturn(5);
+        Channel.GroupListener mockListener = mock(Channel.GroupListener.class);
+        AppCenterIngestion mockIngestion = mock(AppCenterIngestion.class);
+        DefaultChannel channel = new DefaultChannel(mock(Context.class), UUIDUtils.randomUUID().toString(), mockPersistence, mockIngestion, mAppCenterHandler);
+        channel.addGroup(TEST_GROUP, 10, mockInterval, MAX_PARALLEL_BATCHES, mockIngestion, mockListener);
+
+        /* Check values. */
+        verify(mAppCenterHandler, never()).postDelayed(any(Runnable.class), anyLong());
+        verify(mockPersistence, times(2)).countLogs(eq(TEST_GROUP));
+    }
+
+    @Test
+    public void setAppSecretIngestionNoEquals() {
+        long mockInterval = 10000;
+
+        /* Mock channel. */
+        Persistence mockPersistence = mock(Persistence.class);
+        when(mockPersistence.countLogs(TEST_GROUP)).thenReturn(0);
+        Channel.GroupListener mockListener = mock(Channel.GroupListener.class);
+        AppCenterIngestion mockIngestion1 = mock(AppCenterIngestion.class);
+        AppCenterIngestion mockIngestion2 = mock(AppCenterIngestion.class);
+        DefaultChannel channel = new DefaultChannel(mock(Context.class), UUIDUtils.randomUUID().toString(), mockPersistence, mockIngestion1, mAppCenterHandler);
+        channel.addGroup(TEST_GROUP, 10, mockInterval, MAX_PARALLEL_BATCHES, mockIngestion2, mockListener);
+        channel.setAppSecret("app-secret");
+
+        /* Check values. */
+        verify(mockIngestion1, never()).sendAsync(anyString(), anyString(), any(UUID.class), any(LogContainer.class), any(ServiceCallback.class));
```

Comment:
We are making sure sendAsync() is never called - but are can it actually be called in this test? Correct me if im missing something.